spec/requests/players_spec.rb

```diff
@@ -0,0 +1,139 @@
+require "rails_helper"
+require "./spec/support/request_helper"
+
+RSpec.describe Player, type: :request do
+  include RequestSpecHelper
+
+  let!(:team) { create(:team) }
+  let!(:player) { create_list(:player, 10, team_id: team.id) }
+
+  let(:team_id) { team.id }
+  let(:player_id) { player.first.id }
+
+  let(:player_params) {
+    {
+        first_name: "James",
+        second_name: "Kariuki",
+        last_name: "Kaguna",
+        nick_name: "jimnah",
+        id_number: "31262778",
+        dob: "1997-09-19",
+        phone_number: "0715739940",
+        team_id: team_id
+    }
+  }
+
+  describe "POST players/create" do
+
+    context "when the request is valid" do
+      before { post "/players/#{team_id}", params: player_params }
+      it "creates a new player" do
+        expect(json["first_name"]).to eq("James")
+      end
+
+      it "returns status code 201" do
+        expect(response).to have_http_status(201)
+      end
+    end
+
+    context "when the request is invalid" do
+      let(:player_params) {
+        {
+            first_name: "",
+            second_name: "Kariuki",
+            last_name: "Kaguna",
+            nick_name: "jimnah",
+            id_number: "31262778",
+            dob: "1997-09-19",
+            phone_number: "0715739940",
+            team_id: player_id
+        }
+      }
+
+      before { post "/players/#{team_id}", params: player_params }
+
+      it "does not create a new player with empty player first name" do
+        expect(json["first_name"]).to eq(["can't be blank"])
+      end
+
+      it "returns status code 422" do
+        expect(response).to have_http_status(422)
+      end
+    end
+  end
+
+  describe "GET /players" do
+
+    context "when the request is valid" do
+      before { get "/players/#{team_id}/#{player_id}" }
+
+      it "returns a hash of one player with 11 elements" do
+        expect(json.size).to eq 11
+      end
+
+      it "returns status code 200" do
+        expect(response).to have_http_status(200)
+      end
+    end
+
+    context "when the request is invalid" do
+      let(:team_id) { 0 }
+      before { get "/players/#{team_id}/#{player_id}" }
+
+      it "returns an error message and status code 400" do
+        expect(response).to have_http_status(400)
+        expect(json["errors"]).to eq("The player does not exist")
```

Comment:
let's make each example test one thing at a time, one example for status code and the other for expected message.